frontend/packages/dev-console/src/components/import/git/GitSection.tsx

```diff
@@ -77,33 +80,104 @@ const GitSection: React.FC<GitSectionProps> = ({
   const { git: gitErrors = {} } = errors;
   const { url: gitUrlError } = gitErrors as FormikErrors<GitData>;
 
+  const handleBuilderImageRecommendation = React.useCallback(
+    async (detectedBuildTypes: DetectedBuildType[]) => {
+      setFieldValue('image.isRecommending', false);
+      if (gitUrlError) {
+        setFieldValue('image.recommended', '');
+        setFieldValue('image.couldNotRecommend', true);
+        return;
+      }
+      const recommendedBuildType = detectedBuildTypes?.find(
+        ({ type: recommended }) => recommended && builderImages.hasOwnProperty(recommended),
+      );
+      if (recommendedBuildType && recommendedBuildType.type) {
+        setFieldValue('image.couldNotRecommend', false);
+        setFieldValue('image.recommended', recommendedBuildType.type);
+      } else {
+        setFieldValue('image.couldNotRecommend', true);
+        setFieldValue('image.recommended', '');
+      }
+    },
+    [builderImages, gitUrlError, setFieldValue],
+  );
+
+  const handleDevfileStrategyDetecion = React.useCallback(
+    async (gitService: BaseService) => {
+      if (!values.devfile?.devfileSourceUrl) {
+        // No need to check the existence of the file, waste of a call to the gitService for this need
+        const devfileContents = gitService && (await gitService.getDevfileContent());
+        if (!devfileContents) {
+          setFieldValue('devfile.devfileContent', null);
+          setFieldValue('devfile.devfileHasError', true);
+        } else {
+          setFieldValue('devfile.devfileContent', devfileContents);
+          setFieldValue('devfile.devfileHasError', false);
+        }
+      }
+    },
+    [setFieldValue, values.devfile],
+  );
+
   const handleGitUrlChange = React.useCallback(
     async (url: string, ref: string, dir: string) => {
+      setValidated(ValidatedOptions.default);
```

Comment:
Return if form isSubmitting. This is causing re validation when submit is clicked.